### PR TITLE
feat(donut-chart): new module

### LINKED ISSUE #1956: data-visualization: donut chart
This feature will add a new donut chart, the legend, and other CSS portions needed for custom data viz components.

### PR DESCRIPTION
**Pull request by mikehobi**
## Note

Github wouldn't let me reopen/change base for the old PR, here is the old PR with some discussions around changes:

https://github.com/eBay/skin/pull/2326

---

Fixes #1956

- [x] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
This PR adds `donut-chart` and `chart-legend` styling and documentation

## Notes
Our implementation in eBayUI uses Highcharts

#### Dark Mode
image

#### Light Mode
image

#### Dark Mode
image

#### Light Mode
image

## Checklist

- [x] I verify the build is in a non-broken state
- [x] I verify all changes are within scope of the linked issue

- [x] I regenerated all CSS files under dist folder
- [ ] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [x] I tested the UI in dark mode and RTL mode
- [x] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -158,5 +158,8 @@
     "lint-staged": {
         "*.{js,css,md,less,json}": "prettier --write",
         "*.{css,less}": "stylelint --config .stylelintrc"
+    },
+    "overrides": {
+        "clean-css": "^5.3.3"
```
Comment (mikehobi) [inline package.json:163]: @ArtBlue this was the update that allowed me to use container queries

Comment (ArtBlue) [reply package.json:163]: Thanks, @mikehobi !!

File: src/sass/chart-legend/chart-legend.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        var(--color-data-viz-chart-quaternary-background) 3px,
+        var(--color-data-viz-chart-quaternary-background) 4px
+    );
+    border: 1px solid var(--color-data-viz-chart-quaternary-stroke);
+}
+
+.chart-legend dl > .chart-legend__item:nth-child(5)::before {
+    background: var(--color-data-viz-chart-quinary-background);
+    border-color: var(--color-data-viz-chart-quinary-stroke);
+}
```
Comment (agliga) [inline src/sass/chart-legend/chart-legend.scss:70]: If there are more than 5, does it break or not show anything? Maybe we should have a check to default it if theres more than 5.

**Review by agliga — COMMENTED** [review]
LGTM overall, one comment but I think its good to go.

File: Gemfile.lock [hunk 3]
```diff
@@ -282,6 +282,7 @@ GEM
 PLATFORMS
   arm64-darwin-21
   arm64-darwin-22
+  arm64-darwin-23
```
[the remark this answers is not in the document]
Comment (mikehobi) [reply Gemfile.lock:285]: I'm not sure, this was auto generated when I built the docs

File: src/sass/chart-legend/chart-legend.scss (hunk 2 above)
Comment (mikehobi) [reply src/sass/chart-legend/chart-legend.scss:70]: @agliga at the moment, our documentation says no more than 5 data points for any data visualization. So this fits our ebay needs, as for others, they'd have to extend it accordingly

**Review by agliga — APPROVED** [review]

**Comment by ArtBlue** [comment]
Hey @mikehobi , can you resolve the conflicts? After that we can merge this.

**Comment by mikehobi** [comment]
@ArtBlue done!

**Review by ArtBlue — APPROVED** [review]

**Pull request merged**